src/store/configure-store.js

```diff
@@ -0,0 +1,18 @@
+import {createStore, applyMiddleware } from "redux";
+import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
+import thunk from "redux-thunk";
+import reducer from "../reducers";
+
+const Store = preloadedState => {
+  const middlewares = applyMiddleware(thunk);
```

Comment:
Good work, please what does this middleware do in relation to redux?

Reply:
So, every action to be dispatched passes through this middleware. Thunk tells redux if to execute actions as functions or normal actions. We would be needing it in cases where we are making api calls. Our actions will not be plain objects anymore but executable functions